File: medico/Services/Medico.Api/ClientApp/src/app/_models/patientChartNodeAttributes.ts
export class PatientChartNodeAttributes {
    order: number;
    isActive: boolean;
    isNotShownInReport: boolean;
    signedOffOnly: boolean;
    isPredefined: boolean;
    nodeSpecificAttributes: any;

    static createPatientChartNodeAttributes(order: number,
        isActive: boolean, isNotShownInReport: boolean,
        signedOffOnly: boolean, isPredefined: boolean, nodeSpecificAttributes: any = null): PatientChartNodeAttributes {
        const patientChartNodeAttributes =
            new PatientChartNodeAttributes();

        patientChartNodeAttributes.order = order;
        patientChartNodeAttributes.isPredefined = isPredefined;
        patientChartNodeAttributes.isActive = isActive;
        patientChartNodeAttributes.isNotShownInReport = isNotShownInReport;
        patientChartNodeAttributes.signedOffOnly = signedOffOnly;

        patientChartNodeAttributes.nodeSpecificAttributes =
            nodeSpecificAttributes || {};

        return patientChartNodeAttributes;
    }
}

File: medico/Application/Medico.Application/Services/PatientChart/PatientChartNodeAttributes.cs
namespace Medico.Application.Services.PatientChart
{
    public class PatientChartNodeAttributes
    {
        public int Order { get; set; }

        public bool IsActive { get; set; }

        public bool IsPredefined { get; set; }

        public bool IsNotShownInReport { get; set; }

        public bool SignedOffOnly { get; set; }

        public NodeSpecificAttributes NodeSpecificAttributes { get; set; }
    }
}